Sources/XCTest/Public/XCTAssert.swift

```diff
@@ -421,3 +423,14 @@ public func XCTAssertThrowsError<T>(_ expression: @autoclosure () throws -> T, _
         }
     }
 }
+
+public func XCTAssertNoThrow<T>(_ expression: @autoclosure () throws -> T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
+    _XCTEvaluateAssertion(.noThrow, message: message, file: file, line: line) {
+        do {
+             _ = try expression()
+            return .success
+        } catch _ {
+            return .expectedFailure("threw error")
```

Comment:
Ah yes, I see.

I think the failure is terser for `XCTAssertThrowsError` because the error being reported is the *lack of an error*, when an error was expected, so there really isn't more to report. In the case of this new function, we can and should provide whatever details we can about the unexpected error that was encountered.